### PR TITLE
Refactor/abc condition

### PR DESCRIPTION
**Pull request by luke-hill**
## Summary

Fix up some rubocop todo's

## Details

Refactor some complex methods. Use better names where appropriate

## Motivation and Context

Fixes up todo file in rubocop_todo.yml

## How Has This Been Tested?

CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: lib/cucumber/rails/capybara/javascript_emulation.rb [hunk 1]
```diff
@@ -48,27 +48,38 @@ def js_form(document, action, emulated_method, method = 'POST')
...
-          # with non-GET requests
-          if csrf? && !emulated_method.casecmp('get').zero?
-            input = document.create_element('input')
-            input['type'] = 'hidden'
-            input['name'] = csrf_param
-            input['value'] = csrf_token
-            js_form.add_child(input)
-          end
+          # rails will wipe the session if the CSRF token is not sent with non-GET requests
+          add_hidden_csrf_input(document, js_form) if csrf_and_non_get?(emulated_method)
```
[the remark this answers is not in the document]
Comment (luke-hill) [reply lib/cucumber/rails/capybara/javascript_emulation.rb]: I'm not sure what else exists in cucumber-rails. But I try have a single boolean in ifs/whens/unlesses where possible.

I know in a few big projects they use things like `find_or_create` or `update_then_destroy` e.t.c.

I'm sorta half against this. I've fixed up all the other things. I can change this if you want, but I kinda think this is better.

File: features/support/cucumber_rails_helper.rb [hunk 2]
```diff
@@ -2,38 +2,26 @@
...
-    options[:name] ||= 'test_app'
-    command_result =
-      run_command "bundle exec rails new #{options[:name]} --skip-bundle --skip-test-unit --skip-spring --skip-bootsnap #{options[:args]}"
-    expect(command_result).to have_output(/README/)
-    expect(last_command_started).to be_successfully_executed
-    cd options[:name]
-    delete_environment_variable 'RUBYOPT'
-    delete_environment_variable 'BUNDLE_BIN_PATH'
-    delete_environment_variable 'BUNDLE_GEMFILE'
+    validate_rails_new_success(run_rails_new_command(options))
```
Comment (mvz) [inline features/support/cucumber_rails_helper.rb]: I actually meant merging the actual two methods :smile:

Comment (luke-hill) [reply features/support/cucumber_rails_helper.rb]: Ah whoops. I had to uncouple them to reduce the ABC score. This method was around 19, and I think now it scores around 10 or 11. If I put them back together, the score goes back over 15 :(

We could maybe alter the metric. But I'm not sure we should?

File: lib/cucumber/rails/capybara/javascript_emulation.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (mvz) [reply lib/cucumber/rails/capybara/javascript_emulation.rb]: Ok, I'm going to give one more argument and if you're still not convinced you can leave it as-is. :slightly_smiling_face: 

`find_or_create` and `update_then_destroy` combine verbs that both act on the same object. However, `csrf?` does not act on `emulated_method`, whereas `csrf_and_non_get?(emulated_method)` suggests that being `csrf?` and being `non_get?` are both properties of `emuluated_method`.

File: features/support/cucumber_rails_helper.rb (hunk 2 above)
Comment (mvz) [reply features/support/cucumber_rails_helper.rb]: > If I put them back together, the score goes back over 15 :(

:cry: 

Let's leave like this then :man_shrugging:.

File: lib/cucumber/rails/capybara/javascript_emulation.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (luke-hill) [reply lib/cucumber/rails/capybara/javascript_emulation.rb]: That's a fair point. Well explained.

**Pull request merged**